frontend/src/component/pages/shared/UserShortLinksSection.spec.tsx

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+
+import { UserShortLinksSection } from './UserShortLinksSection';
+import { fireEvent, render } from '@testing-library/react';
+import { IPagedShortLinks } from '../../../service/ShortLink.service';
+
+describe('UserShortLinksSection component', () => {
+  test('should render without crash', () => {
+    render(<UserShortLinksSection onPageLoad={jest.fn} />);
+  });
+
+  test('should render nothing when no data is sent', () => {
+    const { container } = render(
+      <UserShortLinksSection onPageLoad={jest.fn} />
+    );
+
+    expect(container.textContent).not.toContain('Created Short Links');
+    expect(container.textContent).not.toContain('Long URL');
+    expect(container.textContent).not.toContain('Alias');
+  });
+
+  test('should render nothing when "total" attribute in urlData is zero', () => {
+    const urlData: IPagedShortLinks = {
+      shortLinks: [],
+      totalCount: 0
+    };
+
+    const { container } = render(
+      <UserShortLinksSection onPageLoad={jest.fn} pagedShortLinks={urlData} />
+    );
+
+    expect(container.textContent).not.toContain('Created Short Links');
+    expect(container.textContent).not.toContain('Long URL');
+    expect(container.textContent).not.toContain('Alias');
+  });
+
+  test('should render when "total" attribute in urlData is non zero with empty urls', () => {
```

Comment:
Huh. I wonder when does this happen.

Reply:
This could happen when the user deletes the last shortLink in the last page. The component might query for the contents for that page which will return zero shortLinks with new total. The pageControl will then take care of adjusting the lastPage and the component queries for the new last page.